appimagebuilder/app_dir/deploy/pacman/venv.py

```diff
@@ -78,51 +81,61 @@ def retrieve(self, packages: [str], excluded_packages: [str] = None):
         )
-        self._logger.debug(command)
-        output = subprocess.run(command, shell=True)
-        self._assert_successful_output(output)
-        return output
+        self._run_command(command, file=file, target=target)
+
+    def _check_deps(self):
+        """
+        Iterates through all items in the list of dependencies
+        defined in the global constant, DEPENDS_ON, and append
+        to the the variable, _deps[dep]
+        :return: None
+        """
+        # check if we have all the deps
+        for dep in DEPENDS_ON:
+            self._deps[dep] = shutil.which(dep)
+            if self._deps[dep] is None:
+                # shutil.which returns None if the executable
+                # was not found on PATH
+                raise FileNotFoundError(
+                    "Could not find '{exe}' on $PATH.")
 
     def _run_pacman_download_packages(self, packages_str, exclude_str):
-        command = "pacman --config %s -Sy --downloadonly --noconfirm %s %s" % (
-            self._config_path,
-            exclude_str,
-            packages_str,
+        self._run_command(
+            "{fakeroot} {pacman} --config {config} -Sy --downloadonly "
+            "--noconfirm {exclude} {packages}",
+            exclude=exclude_str,
+            packages=packages_str
         )
-        self._logger.debug(command)
-        output = subprocess.run(command, shell=True)
-        self._assert_successful_output(output)
-        return output
 
     def _run_pacman_list_packages_and_versions(self, packages_str):
-        command = "pacman --config %s -S --print-format '%%n=%%v' --noconfirm %s" % (
-            self._config_path,
-            packages_str,
-        )
-        self._logger.debug(command)
-        output = subprocess.run(command, stdout=subprocess.PIPE, shell=True)
-        self._assert_successful_output(output)
+        command = \
+            "{pacman} --config {config} -S " \
+            "--print-format '%n=%v' " \
+            "--noconfirm {packages}"
+        output = self._run_command(command, packages=packages_str,
+                                   stdout=subprocess.PIPE)  # noqa:
 
-        return output.stdout.decode("utf-8").splitlines()
+        return output.stdout.read().decode("utf-8").splitlines()  # noqa:
 
     def _run_pacman_list_package_files(self, exclude_str, packages_str):
-        command = "pacman --config %s -S --print-format '%%l' --noconfirm %s %s" % (
-            self._config_path,
-            exclude_str,
-            packages_str,
-        )
+        command = \
+            "{pacman} --config {config} -S " \
+            "--print-format '%%l' " \
+            "--noconfirm {exclude} {packages}"
         self._logger.debug(command)
-        output = subprocess.run(command, stdout=subprocess.PIPE, shell=True)
-        self._assert_successful_output(output)
-
-        files = re.findall("file://(.*)", output.stdout.decode("utf-8"))
+        print(exclude_str, packages_str)
```

Comment:
Is this line required? If so wouldn't it be better to use the logger ?

Reply:
oops, sorry :sweat_smile:

Reply:
Okie, fixed in https://github.com/AppImageCrafters/appimage-builder/pull/67/commits/7111d032aee8b9d2ad4e25dcd6b8ec6250d4cad2